### PR TITLE
Feature enroll class

### PR DESCRIPTION
**Pull request by nilshah98**
Closed #83 
Reopening to activate preview build @komalahire

### CODE REVIEW

**Comment by nilshah98** [comment]
Hi, @komalahire @komalng  are these changes building correctly ?
I can see some babel error while building it on AWS

File: src/components/Class/redux/action.js [hunk 1]
```diff
@@ -42,4 +42,12 @@ export const actions = {
       data,
     };
   },
+  dropOutClass(data, index) {
+    let item = [...data][index];
+    item.enrolled = false;
```
Comment (tarun-dugar) [inline src/components/Class/redux/action.js]: this should not happen here, modification of data should be done in the reducer. The action should just return the id of the class from this action.

File: src/components/Class/redux/reducer.js [hunk 2]
```diff
@@ -56,6 +56,15 @@ export default (state = initialState, action) => {
           data: action.data,
         },
       };
+    case types.GET_UPDATED_DROP_OUT_CLASSES:
+      return {
```
Comment (tarun-dugar) [inline src/components/Class/redux/reducer.js:67]: do the data modification here by accessing the data from state

File: src/components/Class/ClassCard/index.js [hunk 3]
```diff
@@ -83,20 +78,10 @@ function ClassCard({ item, index, handleDropOutData, editClass }) {
...
-          `Something went wrong with error status: ${error.response.status} ${error.response.data.message}`,
-          {
-            position: toast.POSITION.BOTTOM_RIGHT,
-          }
-        );
-      }
-    );
+    }).then(() => {
+      notify();
+      dispatch(classActions.deleteClass(data, id));
```
Comment (tarun-dugar) [inline src/components/Class/ClassCard/index.js]: we should not pass data as part of the action, it's already there in the reducer.

File: src/components/Class/ClassCard/index.js [hunk 4]
```diff
@@ -121,20 +119,10 @@ function ClassCard({ item, handleEnrolledData, handleDropOutData, editClass }) {
...
-            `Something went wrong with error status: ${error.response.status} ${error.response.data.message}`,
-            {
-              position: toast.POSITION.BOTTOM_RIGHT,
-            }
-          );
-        }
-      );
+      .then(() => {
+        notify();
+        dispatch(classActions.enrolledClass(data, index));
```
Comment (tarun-dugar) [inline src/components/Class/ClassCard/index.js]: same feedbacks as in the delete commit

Comment (komalng) [reply src/components/Class/ClassCard/index.js]: https://github.com/navgurukul/bhanwari-devi/pull/85/commits/0fe71dd2f5ca43cae07346e91c0c3558e27cf377

File: src/components/Class/redux/action.js [hunk 5]
```diff
@@ -42,4 +42,12 @@ export const actions = {
       data,
     };
   },
+  dropOutClass(data, index) {
+    let item = [...data][index];
```
[the remark this answers is not in the document]
Comment (komalng) [reply src/components/Class/redux/action.js]: https://github.com/navgurukul/bhanwari-devi/pull/85/commits/9feab343ff61492a0305dafdf53c22262ccd9e11

File: src/components/Class/redux/action.js (hunk 1 above)
Comment (komalng) [reply src/components/Class/redux/action.js]: https://github.com/navgurukul/bhanwari-devi/pull/85/commits/9feab343ff61492a0305dafdf53c22262ccd9e11

File: src/components/Class/redux/reducer.js (hunk 2 above)
Comment (komalng) [reply src/components/Class/redux/reducer.js:67]: https://github.com/navgurukul/bhanwari-devi/pull/85/commits/9feab343ff61492a0305dafdf53c22262ccd9e11

File: src/components/Class/ClassCard/index.js (hunk 3 above)
Comment (komalng) [reply src/components/Class/ClassCard/index.js]: https://github.com/navgurukul/bhanwari-devi/pull/85/commits/9feab343ff61492a0305dafdf53c22262ccd9e11

**Review by tarun-dugar — APPROVED** [review]

**Comment by nilshah98** [comment]
Weirdly this is not building successfully on AWS.
Below are the log output - 
```
                                 # Starting phase: preBuild
                                 # Executing command: yarn install
-react-router@6.8.0" has unmet peer dependency "redux@^3.6.0 || ^4.0.0".
                                    warning " > connected-react-router@6.8.0" has unmet peer dependency "seamless-immutable@^7.1.3".
2021-05-19T07:17:59.872Z [WARNING]: warning " > react-markdown@5.0.3" has unmet peer dependency "@types/react@>=16".
2021-05-19T07:17:59.872Z [WARNING]: warning " > react-redux@7.2.1" has unmet peer dependency "redux@^2.0.0 || ^3.0.0 || ^4.0.0-0".
2021-05-19T07:17:59.874Z [WARNING]: warning "react-scripts > @typescript-eslint/eslint-plugin > tsutils@3.17.1" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
2021-05-19T07:17:59.879Z [WARNING]: warning " > react-youtube@7.13.0" has unmet peer dependency "@types/youtube@^0.0.38".
2021-05-19T07:17:59.879Z [WARNING]: warning " > redux-devtools-extension@2.13.8" has unmet peer dependency "redux@^3.1.0 || ^4.0.0".
2021-05-19T07:18:24.524Z [INFO]: [4/4] Building fresh packages...
2021-05-19T07:18:26.239Z [INFO]: Done in 49.27s.
2021-05-19T07:18:26.272Z [INFO]: # Completed phase: preBuild
                                 # Starting phase: build
2021-05-19T07:18:26.272Z [INFO]: # Executing command: yarn run build
2021-05-19T07:18:26.391Z [INFO]: yarn run v1.22.0
2021-05-19T07:18:26.418Z [INFO]: $ react-scripts build
2021-05-19T07:18:27.173Z [INFO]: Creating an optimized production build...
2021-05-19T07:18:28.076Z [WARNING]: Browserslist: caniuse-lite is outdated. Please run:
                                    npx browserslist@latest --update-db
                                    Why you should do it regularly:
                                    https://github.com/browserslist/browserslist#browsers-data-updating
2021-05-19T07:18:28.297Z [INFO]: Failed to compile.
2021-05-19T07:18:28.297Z [INFO]: ./src/index.js
                                 TypeError: [BABEL] /codebuild/output/src723667495/src/bhanwari-devi/src/index.js: api.assumption is not a function (While processing: "/codebuild/output/src723667495/src/bhanwari-devi/node_modules/babel-preset-react-app/index.js$0$0")
                                 at Generator.next (<anonymous>)
                                 at Generator.next (<anonymous>)
2021-05-19T07:18:28.313Z [WARNING]: error Command failed with exit code 1.
2021-05-19T07:18:28.313Z [INFO]: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2021-05-19T07:18:28.321Z [ERROR]: !!! Build failed
2021-05-19T07:18:28.321Z [ERROR]: !!! Non-Zero Exit Code detected
2021-05-19T07:18:28.322Z [INFO]: # Starting environment caching...
2021-05-19T07:18:28.322Z [INFO]: # Uploading environment cache artifact...
2021-05-19T07:18:28.410Z [INFO]: # Environment caching completed
Terminating logging...
```
Any idea @tarun-dugar

**Comment by nilshah98** [comment]
It is something related to AWS-Amplify and babel caching I think.
I deleted and redployed the branch separately, it worked -
https://feature-enrollclass.dkchei85ij0cu.amplifyapp.com/

**Comment by komalng** [comment]
> It is something related to AWS-Amplify and babel caching I think.
> I deleted and redployed the branch separately, it worked -
> https://feature-enrollclass.dkchei85ij0cu.amplifyapp.com/

**Pull request merged**